portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -74,16 +82,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
         Entity commentEntity = new Entity("Comment");
         commentEntity.setProperty("name", getParameter(request, "name").orElse("Anonymous"));
-        commentEntity.setProperty("email", getParameter(request, "email").orElse("anonymous"));
         commentEntity.setProperty("message", getParameter(request, "message").orElse(""));
         commentEntity.setProperty("timestamp", timestamp);
         commentEntity.setProperty("numLikes", 0);
 
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         datastore.put(commentEntity);
     
-        response.sendRedirect("/index.html");
+        response.sendRedirect("/comments.html");
+
+    }
 
+    /**
+    * Gets the sort-type parameter to determine what order to display comments. 
+    * @return String, if the user input was valid it returns the parameter, otherwise it returns "newest"
+    */
+    private String getSortTypeParam(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String sortType = getParameter(request, "sort-type").orElse("newest").toLowerCase();
+
+        if(!sortType.equals("newest") && !sortType.equals("oldest") && !sortType.equals("popular")) {
```

Comment:
Suggest using an array and checking if the value exists in the array. This will be more maintainable as you add more sort options.

See https://mkyong.com/java/java-check-if-array-contains-a-certain-value/ for some tips